modules/homa-lite/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+		LiquidCurrencyNotIssuedForThisBatch,
+		/// The relaychain's stash account have not been set.
+		RelaychainStashAccountNotSet,
+		/// The total issuance for the Staking currency must be more than zero.
+		InvalidStakedCurrencyTotalIssuance,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	#[pallet::metadata(Balance = "Balance", T::AccountId = "AccountId")]
+	pub enum Event<T: Config> {
+		/// The user has requested some Staking currency to be used to mint Liquid Currency. [batch,
+		/// user, amount]
+		MintRequested(EraIndex, T::AccountId, Balance),
+
+		/// The current batch has been processed. Mint requests can now be completed. [batch,
+		/// staking_total_issuance, liquid_total_issuance]
+		BatchProcessed(EraIndex, Balance, Balance),
+
+		/// The user has claimed some Liquid Currency. [batch, user, amount]
+		LiquidCurrencyClaimed(EraIndex, T::AccountId, Balance),
+
+		/// The relay-chain's stash account ID has been updated.
+		RelaychainStashAccountUpdated(T::AccountId),
+	}
+
+	/// Stores the amount of Staking currency the user has exchanged.
+	/// PendingAmount: double_map: (batch: EraIndex, user: T::AccountId) -> amount: Balance
+	#[pallet::storage]
+	#[pallet::getter(fn pending_amount)]
+	pub type PendingAmount<T: Config> =
+		StorageDoubleMap<_, Blake2_128Concat, EraIndex, Blake2_128Concat, T::AccountId, Balance, ValueQuery>;
+
+	/// The total issuance info for each batch. Used to calculate Staking to Liquid exchange rate.
+	/// BatchTotalIssuanceInfo: map: batch: EraIndex -> batch_total: TotalIssuanceInfo
+	#[pallet::storage]
+	#[pallet::getter(fn batch_total_issuance_info)]
+	pub type BatchTotalIssuanceInfo<T: Config> =
+		StorageMap<_, Blake2_128Concat, EraIndex, TotalIssuanceInfo, OptionQuery>;
+
+	/// The batch that is currency active
+	/// CurrentBatch: value: batch: EraIndex
+	#[pallet::storage]
+	#[pallet::getter(fn current_batch)]
+	pub type CurrentBatch<T: Config> = StorageValue<_, EraIndex, ValueQuery>;
+
+	/// The account in which the staking currency goes into to be transferred to the Relay chain.
+	/// RelaychainStashAccount: value: stash_account: AccountId
+	#[pallet::storage]
+	#[pallet::getter(fn relaychain_stash_account)]
+	pub type RelaychainStashAccount<T: Config> = StorageValue<_, T::AccountId, OptionQuery>;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Request to mint some Liquid currency, by locking up the given amount of Staking
+		/// currency. The exchange does not happen immediately, but on when the batch is processed
+		/// The user then needs to manually claim the Liquid currency once it is ready.
+		///
+		/// Parameters:
+		/// - `amount`: The amount of Staking currency to be exchanged.
+		//#[pallet::weight(< T as Config >::WeightInfo::request_mint())]
+		#[pallet::weight(0)]
+		#[transactional]
+		pub fn request_mint(origin: OriginFor<T>, amount: Balance) -> DispatchResultWithPostInfo {
```

Comment:
`DispatchResult` should be enough as there is no post dispatch info. Same to other dispatchable calls.